flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/tasks/SubtaskCheckpointCoordinatorImpl.java

```diff
@@ -563,24 +563,19 @@ private void finishAndReportAsync(
                         env,
                         asyncExceptionHandler,
                         isRunning);
-        registerConsumer().accept(asyncCheckpointRunnable);
+
+        try {
```

Comment:
The `try-catch` is unnecessary here. You can safely add the IOException to the method signature. I think the only reason for wrapping it before was to match the `Consumer` interface.

Reply:
Ah, sorry for ignore this, I have removed the exception wrapper~